ruff_lsp/server.py

```diff
@@ -1002,7 +1007,27 @@ def _get_settings_by_document(document: workspace.Document | None) -> WorkspaceS
 ###
 
 
-def _executable_path(settings: WorkspaceSettings) -> str:
+def _find_ruff_binary(
+    settings: WorkspaceSettings, version_requirement: SpecifierSet
+) -> str:
+    """Returns the path to the executable, checking that is has the required minimum
+    version."""
+    executable = _find_ruff_binary_path(settings)
+
+    # This call is cached so it should generally be free
+    version = _executable_version(executable)
+    if not version_requirement.contains(version, prereleases=True):
+        # Put the versions first because the pop-up is small
+        message = (
+            f"ruff {version_requirement} required, but {version} found at {executable}"
+        )
+        show_error(message)
```

Comment:
Did this end up working?

Reply:
yep, see screenshot in the PR description